app/src/main/java/org/fitchfamily/android/dejavu/BackendService.kt

```diff
@@ -735,16 +750,17 @@ class BackendService : LocationBackendService() {
             }
         }
 
-        // Estimate location using weighted average of the most recent
-        // observations from the set of RF emitters we have seen. We cull
-        // the locations based on distance from each other to reduce the
-        // chance that a moved/moving emitter will be used in the computation.
-        val weightedAverageLocation = when (cull) {
-            1 -> locations.medianCullSafe()
-            2 -> locations.weightedAverage()
-            else -> culledEmitters(locations)?.weightedAverage()
-        }
-        if (weightedAverageLocation != null && notNullIsland(weightedAverageLocation)) {
+        if (!locations.isEmpty()) {
```

Comment:
I'd rather have a simple `if (locations.isEmpty()) return` here, reduces unnecessary indent below and works just as well.

Reply:
Caused by maxOf on an empty list
```
FATAL EXCEPTION: DefaultDispatcher-worker-4
Process: helium314.localbackend, PID: 15459
java.util.NoSuchElementException
	at org.fitchfamily.android.dejavu.UtilKt.location(Util.kt:237)
	at org.fitchfamily.android.dejavu.UtilKt.weightedAverage(Util.kt:229)
	at org.fitchfamily.android.dejavu.BackendService.endOfPeriodProcessing(BackendService.kt:750)
	at org.fitchfamily.android.dejavu.BackendService.access$endOfPeriodProcessing(BackendService.kt:46)
	at org.fitchfamily.android.dejavu.BackendService$startProcessingPeriodIfNecessary$1$1.invokeSuspend(BackendService.kt:685)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

Reply:
I know, there is only this `maxOf` where the crash can occur. I just meant that it's easier to `return` if the list is empty, as nothing below this line needs is called if there is no location.